scripts/functions

```diff
@@ -1305,6 +1305,24 @@ CT_DoBuildTargetTuple() {
     CT_ARCH_TARGET_LDFLAGS="${CT_ARCH_TARGET_LDFLAGS} ${CT_ARCH_ENDIAN_LDFLAG}"
 }
 
+# This function determines the target tuple for a given set of compiler
+# flags, using either GCC's multiarch feature (if supported) or using
+# fallback architecture-specific functions.
+CT_DoMultilibTarget() {
+    local multi_flags="$1"
+    local target="$2"
```

Comment:
In fact, if you pass the target as first argument, you can then pass all
multi flags ar extra args and store them in an array:

    CT_DoMultilibTarget() {
        local target="${1}"; shift
        local -a multi_flags=( "${@}" )
        local gcc_multiarch
    
        gcc_multiarch="$( "${CT_TARGET}-gcc" -print-multiarch "${multi_flags[@]}" )"
        if [ -n "${gcc_multiarch=}" ]; then
            printf "%s\n" "${gcc_multiarch=}"
        fi

        CT_DoArchMultilibTarget "${target}" "${multi_flags[@]}"
    }

(of course, propagate the target <-> multi_flags switch to the previous
commit, that introduces `CT_DoArchMultilibTarget`.)

(and the reason I like to have spaces after `(` and before `)` is for
readability).